src/WLR/Types/DamageRing.hsc

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE PatternSynonyms #-}
+module WLR.Types.DamageRing where
+
+#define WLR_USE_UNSTABLE
+#include <wlr/types/wlr_damage_ring.h>
+
+import Foreign.Ptr (Ptr, plusPtr)
+import Foreign (Storable(..), Int32, peekArray, pokeArray)
+import Foreign.C.Types (CInt(..), CBool(..), CSize)
+
+import WLR.Util.Box (WLR_box)
+import PIXMAN.Pixman (PIXMAN_region32)
+
+-- | For triple buffering, a history of two frames is required.
+pattern WLR_DAMAGE_RING_PREVIOUS_LEN :: (Eq a, Num a) => a
+pattern WLR_DAMAGE_RING_PREVIOUS_LEN = 2
+
+-- | current - Difference between the current buffer and the previous one
+-- | previous and previous_idx are private state
+{{ struct wlr/types/wlr_damage_ring.h,
+    wlr_damage_ring,
+    width, CInt,
+    height, CInt,
```

Comment:
These should be `Int32`, I believe.

Reply:
Ugh, thank you. I check https://hackage.haskell.org/package/base-4.19.1.0/docs/Foreign-C-Types.html#g:5 and I see "CInt Int32" and think that's what I'm supposed to use.

I need to remember to specify the concrete type, if that's what the C lib specifies.